### ISSUE TITLE
`enforce-named-default`

### ISSUE THREAD

**Issue by paulshryock** [import-js/eslint-plugin-import#2174]
Is there a rule to do the opposite of [no-named-default](https://github.com/import-js/eslint-plugin-import/blob/master/docs/rules/no-named-default.md)? I would like to enforce only named defaults in my module.

Example:
```javascript
import { default as myModule } from 'my-module' // passes
import myModule from 'my-module' // fails
```

**Comment by ljharb** [comment]
Why would you want that? Defaults aren't the same as named; wouldn't you want to prohibit default exports entirely in that case?

**Comment by paulshryock** [comment]
> Why would you want that? Defaults aren't the same as named; wouldn't you want to prohibit default exports entirely in that case?

No, I'd like to allow default exports, but I'd prefer a unified syntax. So all my imports would be inside curly braces, for consistency.

**Comment by ljharb** [comment]
That wouldn’t be consistent though - it would only appear so on the surface. Default and named imports are fundamentally distinct things, and the consistent thing is to have them use distinct syntax.

**Comment by paulshryock** [comment]
Where it's different is when I'm importing a cjs module into an esm module:

```javascript
import myModule from 'my-module'
```

`myModule` in this case is an object with a `default` property:

```javascript
{
  default: ...
}
```

So if it was a function, for example, instead of calling `myModule()`, I need to call `myModule.default()`. Importing `default` as `myModule` allows me to just call `myModule()`. An opt-in rule to enforce this would just make my life easier in this case.

It's not critical; just thought I'd ask if there was an existing rule I could opt-in to.

**Comment by ljharb** [comment]
In that case, I’d file a bug on the CJS module - packages shouldn’t be exposing Babel implementation details like dot default.

**Comment by paulshryock** [comment]
> In that case, I’d file a bug on the CJS module - packages shouldn’t be exposing Babel implementation details like dot default.

I'm the one writing the cjs module. More specifically, I'm writing an esm module and using esbuild to output both esm and cjs dist files in order to support both types of module consumers. As you say, there's a bug in my implementation as `.default` shouldn't be necessary. My rule suggestion was a temporary workaround, but I know I should prioritize fixing the root problem.
